bundles/auth/core/src/main/java/org/apache/sling/auth/core/impl/SlingAuthenticator.java

```diff
@@ -1657,11 +1657,15 @@ protected AbstractAuthenticationHandlerHolder createHolder(String path,
         }
 
         boolean isNodeRequiresAuthHandler() {
-            if (requestPath.equalsIgnoreCase(handlerPath)) {
+            final String requestPathNoSelectors = requestPath.contains(".") ?
```

Comment:
This condition is not correct. Just that a path contains a dot does not mean that the part before the first dot is the resource path. Basically for the path `/some/path.extension/test/some.other.html`, you cannot tell without knowing the repository structure whether there is a resource at `/some/path.extension/test/some`, which means that there is no suffix and the selector `other` and the extension `html` or in case there is a resource only at `/some/path` the part `extension` would be the extension, there would be no selector and `/test/some.other.html` would be the suffix. See also https://sling.apache.org/documentation/the-sling-engine/url-decomposition.html.